Source/WebCore/rendering/RenderLayerFilters.cpp

```diff
@@ -199,6 +199,13 @@ GraphicsContext* RenderLayerFilters::beginFilterEffect(RenderElement& renderer,
     resetDirtySourceRect();
 
     filter.setFilterRegion(m_filterRegion);
+
+    if (filter.filterMode() == FilterMode::Native) {
+        auto styles = filter.createStyles();
+        context.addFilterEffectStyles(styles);
```

Comment:
This is where you need to save the state, set the style and start a transparency layer for each filter.